src/components/button/index.tsx

```diff
@@ -23,18 +23,62 @@ type SubmitTypeProps = SharedProps & {
 
 type DefaultTypeProps = SharedProps & WithOnClick<'button'>;
 
-export type ButtonProps = ButtonTypeProps | SubmitTypeProps | DefaultTypeProps;
+export type ButtonOnClickProps<AsProp = 'button'> =
```

Comment:
Missed that when playing around but you can probably move the AsProp to the shared button props so we don't need to repeat it.

Reply:
There is another work around here. Thanks!